db/migrate/20200416073135_create_credit_transactions.rb

```diff
@@ -0,0 +1,12 @@
+class CreateCreditTransactions < ActiveRecord::Migration[6.0]
+  def change
+    create_table :credit_transactions do |t|
+      t.integer :credits, default: 0, null: false
+      t.string :reason, default: 'other', null: false
+      t.references :content, polymorphic: true
+      t.belongs_to :user, foreign_key: true
+      t.index :reason
```

Comment:
@akshay1y2 reason need not to be index. Also instead of reason being string make it text column so that we can save long string